broker-core/src/main/java/io/zeebe/broker/system/monitoring/BrokerHealthCheckService.java

```diff
@@ -0,0 +1,118 @@
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package io.zeebe.broker.system.monitoring;
+
+import io.atomix.cluster.MemberId;
+import io.atomix.core.Atomix;
+import io.atomix.protocols.raft.partition.RaftPartitionGroup;
+import io.zeebe.broker.clustering.base.partitions.Partition;
+import io.zeebe.broker.system.SystemContext;
+import io.zeebe.servicecontainer.Injector;
+import io.zeebe.servicecontainer.Service;
+import io.zeebe.servicecontainer.ServiceGroupReference;
+import io.zeebe.servicecontainer.ServiceStartContext;
+import io.zeebe.servicecontainer.ServiceStopContext;
+import io.zeebe.util.sched.Actor;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class BrokerHealthCheckService extends Actor implements Service<Void> {
+  private Map<Integer, Boolean> partitionInstallStatus;
+
+  /* set to true when all partitions are installed. Once set to true, it is never
+  changed. */
+  private volatile boolean brokerStarted = false;
+
+  private final ServiceGroupReference<Partition> leaderInstallReference =
+      ServiceGroupReference.<Partition>create()
+          .onAdd((name, partition) -> updateBrokerReadyStatus(partition.getPartitionId()))
+          .build();
+
+  private final ServiceGroupReference<Partition> followerInstallReference =
+      ServiceGroupReference.<Partition>create()
+          .onAdd((name, partition) -> updateBrokerReadyStatus(partition.getPartitionId()))
+          .build();
+
+  private Atomix atomix;
+  private final Injector<Atomix> atomixInjector = new Injector<>();
+
+  public boolean isBrokerReady() {
+    return brokerStarted;
+  }
+
+  public BrokerHealthCheckService(SystemContext context) {
+    context.getScheduler().submitActor(this);
+  }
+
+  private void updateBrokerReadyStatus(int partitionId) {
+    actor.call(
+        () -> {
+          if (!brokerStarted) {
+            final Map<Integer, Boolean> partitionsInstalled =
+                getOrInitializePartitionInstallStatus();
+            partitionsInstalled.put(partitionId, true);
+            brokerStarted = !partitionsInstalled.containsValue(false);
+          }
+        });
+  }
+
+  private Map<Integer, Boolean> getOrInitializePartitionInstallStatus() {
+    if (partitionInstallStatus == null) {
+      partitionInstallStatus = new HashMap<>();
```

Comment:
I din't do it because Atomix may be not started at the service start. But now when I think about it, there is nothing preventing it from depending on AtomixJoinService.

Reply:
Do we gain anything from starting it before?

Reply:
No. I updated it to depend on AtomixJoinService and initialize on service start.